topic-operator/src/main/java/io/strimzi/operator/topic/TopicOperator.java

```diff
@@ -1234,17 +1234,20 @@ public void addKafkaTopics(List<KafkaTopic> ktList) {
                                 return Future.succeededFuture();
                             } else {
                                 LOGGER.debug("{}: Have private topic for topic {} in Kafka", logContext, topicName);
-                                Future<Void> map = reconcileWithPrivateTopic(logContext, topicName, topic, this)
+                                Promise<Void> map = Promise.promise();
+                                        reconcileWithPrivateTopic(logContext, topicName, topic, this)
                                         .<Void>map(ignored -> {
                                             LOGGER.debug("{} reconcile success -> succeeded", topicName);
                                             succeeded.add(topicName);
+                                            map.complete();
                                             return null;
                                         }).otherwise(error -> {
                                             LOGGER.debug("{} reconcile error -> failed", topicName);
                                             failed.put(topicName, error);
+                                            map.fail(error.getCause());
```

Comment:
Why are we getting the cause here? And what was the outer exception?

Reply:
It is an `OperatorException` with message "Error getting KafkaTopic my-topic during initial kafka my-topic reconciliation"